### PR TITLE
Check that TransformLayer has a finite matrix

### PR DESCRIPTION
**Pull request by liyuqian**
To catch issues like https://github.com/flutter/flutter/issues/30586

We'll also upload a test in the framework that will trigger this DCHECK
if https://github.com/flutter/engine/pull/8467 were reverted.

### CODE REVIEW

File: flow/layers/transform_layer.cc [hunk 1]
```diff
@@ -11,6 +11,17 @@ TransformLayer::TransformLayer() = default;
 TransformLayer::~TransformLayer() = default;
 
 void TransformLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
+  // Checks (in some degree) that SkMatrix transform_ is valid and initialized.
```
Comment (jason-simmons) [inline flow/layers/transform_layer.cc]: Can the `TransformLayer` constructor initialize the `transform_` to the identity matrix?

Another option would be setting a flag when `set_transform` is called and checking that flag here.

File: flow/layers/transform_layer.cc [hunk 2]
```diff
@@ -11,6 +11,17 @@ TransformLayer::TransformLayer() = default;
...
+  //
+  // If transform_ is uninitialized, this assert may look flaky as it doesn't
+  // fail all the time, and some rerun may make it pass. But don't ignore it and
+  // just rerun the test if this is triggered, since even a flaky failure here
+  // may signify a potentially big problem in the code.
+  //
+  // We have to write this flaky test because there is no reliable way to test
+  // whether a variable is initialized or not in C++.
+  FML_DCHECK(transform_.isFinite());
```
Comment (chinmaygarde) [inline flow/layers/transform_layer.cc]: The unopt variant is not used in any serious manner when exercising flow. Having uninitialized stuff in the matrix is quite serious. Maybe just terminate the process here.

File: flow/layers/transform_layer.cc (hunk 1 above)
Comment (liyuqian) [reply flow/layers/transform_layer.cc]: I'll add a flag, together with this isFinite check. The reason that we still need this isFinite check is that someone could call set_transform with an uninitialized SkMatrix.

File: flow/layers/transform_layer.cc (hunk 2 above)
Comment (liyuqian) [reply flow/layers/transform_layer.cc]: I've changed FML_DCHECK to FML_CHECK.

**Comment by liyuqian** [comment]
@chinmaygarde @jason-simmons : how does the new patch look to you?

**Comment by chinmaygarde** [comment]
Instead of the `is_set_` flag, IMO, it would be cleaner to initialize the matrix to identity in the constructor but still do the isFinite check.

**Pull request merged**